docs/how-to-guides.md

```diff
@@ -0,0 +1,383 @@
+# How-to guides 
+
+## Migrate a ReScript library to Melange
+
+It is possible to use existing [ReScript](https://rescript-lang.org/) (formerly
+BuckleScript) code with Melange, mostly as is. However, as both projects evolve
+in different directions over time, it will become more challenging to do so as
+time goes by, as some of the most recent features of ReScript might not be
+directly convertible to make them work with Melange.
+
+For this reason, the recommendation is to migrate libraries at a time where they
+were compatible with past versions of ReScript, for example v9 or at least v10.
+
+These are the steps to follow:
+
+- Add an `opam` file
+- Add a `dune-project` file
+- Replace the `bsconfig.json` file with one or multiple `dune` files
+- (Optional) Migrate from ReScript syntax to Reason or OCaml syntaxes
+- Make sure everything works: `dune build @all`
```

Comment:
`dune build @all` == `dune build`, so we can avoid potentially confusing people who don't know aliases yet

```suggestion
- Make sure everything works: `dune build`
```